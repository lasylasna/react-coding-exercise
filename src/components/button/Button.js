import React from "react";
import PropTypes from "prop-types";

const Button = ({ text, onClick, className, color }) => {
    return (
        <button data-testid="button" className={className} onClick={onClick} style={{ color: color }} >
            {text}{" "}
        </button>
    );
};
Button.defaultProps = {
    color: "rgb(108,212,218)",
};

Button.propTypes = {
    text: PropTypes.string,
    color: PropTypes.string,
    onClick: PropTypes.func,
};

export default Button;
