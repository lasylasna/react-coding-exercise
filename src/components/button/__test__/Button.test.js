import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import Button from '../Button';
import "@testing-library/jest-dom/extend-expect";
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

afterEach(cleanup);

test('renders without crash', () => {
    const div = document.createElement("div");
    render(<Button />, div);
});

test('render button', () => {
    const div = document.createElement("div");
    const { getByTestId } = render(<Button text="submit" />, div);
    expect(getByTestId('button')).toHaveTextContent("submit")
});


//track the snapshots 
test("snapshot test", () => {
    const nodeTree = renderer.create(<Button text="submit" />).toJSON();
    expect(nodeTree).toMatchSnapshot();
})

