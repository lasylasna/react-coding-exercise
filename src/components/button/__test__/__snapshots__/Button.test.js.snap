// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<button
  data-testid="button"
  style={
    Object {
      "color": "rgb(108,212,218)",
    }
  }
>
  submit
   
</button>
`;
